docs/HACKING.md

```diff
@@ -200,3 +200,59 @@ export KUBECONFIG=$XDG_RUNTIME_DIR/kubeconfig
 
 And from there debug the cluster live while the tests are running. Though be
 aware that it will be quickly torn down as soon as the tests succeed or fail.
+
+# Building and installing a custom release image for tesing the MCO
+
+During the MCO development there are certain test scenarios for any of the component
+that would require a cluster running with a custom release to properly test your
+code (i.e. test the interaction of the MCO with the CVO).
+
+In those situations, you will need to:
+
+1) build the MCO components images you're interested into testing
+2) build a custom release payload
+3) install a cluster with your custom release payload
+
+## Build the MCO components images
+
+To build the image for any of the component run:
+
+```
+make image-{component}
+```
+
+`{component}` can be either `operator`, `daemon`, `controller` or `server`.
+Note, quay.io or any other public registry isn't strictly required, you can use a local
+registry for this flow to still work as long as those images are pullable.
+
+After the build is complete, make sure to push the image to a registry (i.e. `quay.io/user/machine-config-{component}`).
+
+## Build a custom release payload
+
+Now that your have your custom component images, to build a custom release payload, run:
+
+```
+oc adm release new -n openshift --server https://api.ci.openshift.org \
+                                --from-image-stream "origin-v4.0" \
+                                --to-image quay.io/use/origin-release:v4.0 \
+                                machine-config-{component}=quay.io/user/machine-config-{component}
```

Comment:
there's a known limitation now in code for this though https://github.com/openshift/machine-config-operator/issues/421, which can be worked around with https://github.com/openshift/machine-config-operator/issues/421 (basically all operands have to be at the very same version or the installation just fails).
That can be worked around by bundling all the components though (see https://github.com/openshift/machine-config-operator/issues/421#issuecomment-463392262)

Reply:
Probably worth just mentioning this limitation and link to the issue directly in the docs.

Reply:
addressed